presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2132,7 +2132,7 @@ public PhysicalOperation visitSemiJoin(SemiJoinNode node, LocalExecutionPlanCont
         public PhysicalOperation visitTableWriter(TableWriterNode node, LocalExecutionPlanContext context)
         {
             // Set table writer count
-            if (node.getPartitioningScheme().isPresent()) {
+            if (node.getPartitioningScheme().isPresent() || stageExecutionDescriptor.isRecoverableGroupedExecution()) {
```

Comment:
Maybe add a comment:
```
TODO: This slows down table write. Once we have TableWriterMergeOperator discussed in #13098, this is no longer required.
```

cc @arhimondr

Reply:
Actually should we guard this with a session property given this slow down the query wall time ? :)

Reply:
I changed the recoverability check for `TableWriterNode` in `PlanFragmenter`. Instead of removing the partitioning check entirely, now it would check the `TASK_WRITER_COUNT` to be 1 if target table does not have partitioning/